### PR TITLE
[frontend] #2020 Backport admin staistics to client app (Part 3)

### PR DESCRIPTION
**Pull request by yrakova**
#2020

**Description**:
* month time tracking updated (select month)
* week time tracking chart added
* info dialog added
* minor fixes
* tests added

### CODE REVIEW

File: frontend/app/components/statistics/bar-chart/styles.css [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#chart {
+    display: flex;
+    justify-content: flex;
```
Comment (lifeart) [inline frontend/app/components/statistics/bar-chart/styles.css:3]: looks like we could use plain tailwind classes:

'flex justify-center' (https://tailwindcss.com/docs/justify-content)

File: frontend/app/components/statistics/info-dialog/styles.css [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  justify-content: center;
+}
+& .info-dialog-actions button {
+  border: 1px solid #7979f2;
+  box-sizing: border-box;
+  box-shadow: 0px 4px 16px #5947b240;
+  border-radius: 16px;
+  background: linear-gradient(90deg, #7979f2 0%, #a179f2 100%);
+  text-transform: uppercase;
+  color: #ffffff;
```
Comment (lifeart) [inline frontend/app/components/statistics/info-dialog/styles.css:44]: same here, let's use tailwind classes for styles

File: frontend/app/components/statistics/week-time-track/styles.css [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/*! purgecss start ignore */
+& .title,
+& .content .empty-data {
+  font-weight: 600;
+  font-size: 12px;
+  line-height: 24px;
```
Comment (lifeart) [inline frontend/app/components/statistics/week-time-track/styles.css:6]: looks like we could style it using tailwind

File: frontend/app/components/ui/close/index.hbs [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<svg
+  width="24"
+  height="24"
```
Comment (lifeart) [inline frontend/app/components/ui/close/index.hbs:3]: looks like we could save it as .svg into public/... folder (if it's not dynamic)

File: frontend/tests/integration/components/statistics/component-test.js [hunk 5]
```diff
@@ -37,4 +37,26 @@ module('Integration | Component | statistics', function (hooks) {
       'getUserStatisticsByYear called on init',
     );
   });
+
+  test('it shows info dialog', async function (assert) {
+    // Set any properties with this.set('myProperty', 'value');
+    // Handle any actions with this.set('myAction', function(val) { ... });
+    const stubNetworkService = Service.extend({
+      getUserStatisticsByYear(from, to) {
```
Comment (lifeart) [inline frontend/tests/integration/components/statistics/component-test.js:45]: let's use mirage to mock BE data

File: frontend/app/components/statistics/bar-chart/component.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { trackedRef } from 'ember-ref-bucket';
+import { bar, bb, Chart } from 'billboard.js';
+import { ChartOptions, Data } from 'billboard.js';
```
Comment (lifeart) [inline frontend/app/components/statistics/bar-chart/component.ts:5]: looks like here is duplicated import, I assume we could import types as-is and importy library lazily (to get lazy chunk of data)

const { ... } = await import('billboard.js');

File: frontend/app/components/statistics/bar-chart/component.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    this.chartColumns = data;
+    this.chart?.load({ columns: this.chartColumns });
+  }
+
+  set options(options: BarOptionsType) {
+    if (!options) {
+      return;
+    }
+    this.chartOptions = options;
+    if (this.chartElemRef) {
```
Comment (lifeart) [inline frontend/app/components/statistics/bar-chart/component.ts:49]: looks like side-effect, could we move it to didUpdateOptions?

File: frontend/app/components/statistics/bar-chart/styles.css (hunk 1 above)
Comment (yrakova) [reply frontend/app/components/statistics/bar-chart/styles.css]: Done

File: frontend/app/components/statistics/info-dialog/styles.css (hunk 2 above)
Comment (yrakova) [reply frontend/app/components/statistics/info-dialog/styles.css]: Done

File: frontend/app/components/statistics/week-time-track/styles.css (hunk 3 above)
Comment (yrakova) [reply frontend/app/components/statistics/week-time-track/styles.css]: Done

File: frontend/app/components/ui/close/index.hbs (hunk 4 above)
Comment (yrakova) [reply frontend/app/components/ui/close/index.hbs]: Done

File: frontend/tests/integration/components/statistics/component-test.js (hunk 5 above)
Comment (yrakova) [reply frontend/tests/integration/components/statistics/component-test.js]: Done

File: frontend/app/components/statistics/bar-chart/component.ts (hunk 6 above)
Comment (yrakova) [reply frontend/app/components/statistics/bar-chart/component.ts]: Done

File: frontend/app/components/statistics/bar-chart/component.ts (hunk 7 above)
Comment (yrakova) [reply frontend/app/components/statistics/bar-chart/component.ts:46]: I've refactored the bar-char component, I believe now it looks better

**Review by lifeart — APPROVED** [review]

**Pull request merged**